client/rest-high-level/src/main/java/org/elasticsearch/client/asyncsearch/SubmitAsyncSearchRequest.java

```diff
@@ -0,0 +1,282 @@
+    }
+
+    /**
+     * Get the search type to execute, defaults to {@link SearchType#DEFAULT}.
+     */
+    public SearchType getSearchType() {
+        return this.searchRequest.searchType();
+    }
+
+    /**
+     * Sets if this request should allow partial results. (If method is not called,
+     * will default to the cluster level setting).
+     */
+    public void setAllowPartialSearchResults(boolean allowPartialSearchResults) {
+        this.searchRequest.allowPartialSearchResults(allowPartialSearchResults);
+    }
+
+    /**
+     * Gets if this request should allow partial results.
+     */
+    public Boolean getAllowPartialSearchResults() {
+        return this.searchRequest.allowPartialSearchResults();
+    }
+
+    /**
+     * Sets the number of shard results that should be reduced at once on the coordinating node. This value should be used as a protection
+     * mechanism to reduce the memory overhead per search request if the potential number of shards in the request can be large.
+     */
+    public void setBatchedReduceSize(int batchedReduceSize) {
+        this.searchRequest.setBatchedReduceSize(batchedReduceSize);
+    }
+
+    /**
+     * Gets the number of shard results that should be reduced at once on the coordinating node.
+     * This defaults to 5 for {@link SubmitAsyncSearchRequest}.
+     */
+    public int getBatchedReduceSize() {
+        return this.searchRequest.getBatchedReduceSize();
+    }
+
+    /**
+     * Sets if this request should use the request cache or not, assuming that it can (for
+     * example, if "now" is used, it will never be cached). By default (not set, or null,
+     * will default to the index level setting if request cache is enabled or not).
+     */
+    public void setRequestCache(Boolean requestCache) {
+        this.searchRequest.requestCache(requestCache);
+    }
+
+    /**
+     * Gets if this request should use the request cache or not.
+     * Defaults to `true` for {@link SubmitAsyncSearchRequest}.
+     */
+    public Boolean getRequestCache() {
+        return this.searchRequest.requestCache();
+    }
+
+    /**
+     * Gets if the source of the {@link SearchSourceBuilder} initially used on this request.
+     */
+    public SearchSourceBuilder getSearchSource() {
+        return this.searchRequest.source();
+    }
+
+    @Override
+    public Optional<ValidationException> validate() {
+        final ValidationException validationException = new ValidationException();
+        if (searchRequest.scroll() != null) {
+            validationException.addValidationError("[scroll] queries are not supported");
+        }
+        if (searchRequest.isSuggestOnly()) {
+            validationException.addValidationError("suggest-only queries are not supported");
+        }
+        if (searchRequest.isCcsMinimizeRoundtrips()) {
+            validationException.addValidationError("[ccs_minimize_roundtrips] is not supported on async search queries");
+        }
```

Comment:
some of these checks become redundant once the search request is hidden. Are we going to be able to remove the getter for the search request, or will it be package protected? I guess that will play a role in whether these values are at least theoretically settable from the same package, or not at all.

Reply:
That getter can probably go, other need to be accessible from the AsyncRequestConverters, which currently need to stay in `org.elasticsearch.client` to work with existing infra in RequestConverters. Moving the former proved to be difficult. I think I can remove the validation checks for "ccs_minimize_roundtrips" and "scroll" though because now they should not be settable any more (and there is an extra layer of protection on the REST layer later.

Reply:
there should be no setters/getters for scroll and `ccs_minimize_roundtrips` anymore now so I removed the validation.